jframework-test/src/test/java/com/github/neatlife/jframework/test/util/IpUtilTest.java

```diff
@@ -0,0 +1,104 @@
+package com.github.neatlife.jframework.test.util;
+
+import com.github.neatlife.jframework.fundation.util.IpUtil;
+import org.junit.Assert;
+import org.junit.Test;
+import org.powermock.api.mockito.PowerMockito;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class IpUtilTest {
+
+    @Test
+    public void testGetIpAddr1() {
+        HttpServletRequest httpServletRequest =
+                PowerMockito.mock(HttpServletRequest.class);
+        PowerMockito.when(httpServletRequest.getHeader("x-forwarded-for"))
+                .thenReturn("0:0:0:0:0:0:0:1");
+
+        Assert.assertEquals("127.0.0.1", IpUtil.getIpAddr(httpServletRequest));
+    }
+
+    @Test
+    public void testGetIpAddr2() {
+        HttpServletRequest httpServletRequest =
+                PowerMockito.mock(HttpServletRequest.class);
+        PowerMockito.when(httpServletRequest.getHeader("Proxy-Client-IP"))
+                .thenReturn("0:0:0:0:0:0:0:1");
+
+        Assert.assertEquals("127.0.0.1", IpUtil.getIpAddr(httpServletRequest));
+    }
+
+    @Test
+    public void testGetIpAddr3() {
+        HttpServletRequest httpServletRequest =
+                PowerMockito.mock(HttpServletRequest.class);
+        PowerMockito.when(httpServletRequest.getHeader("X-Forwarded-For"))
+                .thenReturn("0:0:0:0:0:0:0:1");
+
+        Assert.assertEquals("127.0.0.1", IpUtil.getIpAddr(httpServletRequest));
+    }
+
+    @Test
+    public void testGetIpAddr4() {
+        HttpServletRequest httpServletRequest =
+                PowerMockito.mock(HttpServletRequest.class);
+        PowerMockito.when(httpServletRequest.getHeader("WL-Proxy-Client-IP"))
+                .thenReturn("0:0:0:0:0:0:0:1");
+
+        Assert.assertEquals("127.0.0.1", IpUtil.getIpAddr(httpServletRequest));
+    }
+
+    @Test
+    public void testGetIpAddr5() {
+        HttpServletRequest httpServletRequest =
+                PowerMockito.mock(HttpServletRequest.class);
+        PowerMockito.when(httpServletRequest.getHeader("X-Real-IP"))
+                .thenReturn("0:0:0:0:0:0:0:1");
+
+        Assert.assertEquals("127.0.0.1", IpUtil.getIpAddr(httpServletRequest));
+    }
+
+    @Test
+    public void testGetIpAddr6() {
+        HttpServletRequest httpServletRequest =
+                PowerMockito.mock(HttpServletRequest.class);
+        PowerMockito.when(httpServletRequest.getRemoteAddr())
+                .thenReturn("10.0.0.1");
+
+        Assert.assertEquals("10.0.0.1", IpUtil.getIpAddr(httpServletRequest));
+    }
+
+    @Test
+    public void testInternalIp() {
+        Assert.assertTrue(IpUtil.internalIp("10.0.0.1"));
```

Comment:
You could check for "127.0.0.1", a commonly-known example of an internal ip-address and specifically checked for in the source.

Reply:
Done